pkg/alertmanager/alertmanager.go

```diff
@@ -233,3 +244,67 @@ func (am *Alertmanager) Stop() {
 func (am *Alertmanager) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	am.router.ServeHTTP(w, req)
 }
+
+// buildIntegrationsMap builds a map of name to the list of integration notifiers off of a
+// list of receiver config.
+func buildIntegrationsMap(nc []*config.Receiver, tmpl *template.Template, logger log.Logger) (map[string][]notify.Integration, error) {
+	integrationsMap := make(map[string][]notify.Integration, len(nc))
+	for _, rcv := range nc {
+		integrations, err := buildReceiverIntegrations(rcv, tmpl, logger)
+		if err != nil {
+			return nil, err
+		}
+		integrationsMap[rcv.Name] = integrations
+	}
+	return integrationsMap, err
+}
+
+// buildReceiverIntegrations builds a list of integration notifiers off of a
+// receiver config.
+// Taken from https://github.com/prometheus/alertmanager/blob/94d875f1227b29abece661db1a68c001122d1da5/cmd/alertmanager/main.go#L112-L159.
```

Comment:
We should look into exporting this upstream function if we can.

Reply:
Looking at old alertmanager releases, it was actually exported in some different package. It was un-exported and moved there in recent releases. I will attempt exporting from alertmanager.

Reply:
Tried to export it in upstream by putting it in the right place (`alertmanager/notify`), but running into a lot of cyclic import conflicts (tried to remove most, but hit a dead-end to fix it without duplicating some code). If its fine we could leave it as it is here.